### ISSUE TITLE
Interesting observation about results over epochs

### ISSUE THREAD

**Issue by ersinyar** [fchollet/keras#4868]
I observed that accuracy or the other metrics such as auc become best for a test set at epoch 3 no matter what dataset I use to train an LSTM model, even the example imdb dataset used in the example provided by keras. In other words, I train an LSTM model and search over a grid of different epoch values say from set [3, 5, 10, ...]  and the best value of accuracy or auc for my test set seems to come from a model which is trained over 3 epochs. I also observed this behavior using imdb lstm example of keras. I could not find a reasonable explanation for this result. Has anyone experienced such an unusual behavior?
